### ISSUE TITLE
DockerHub password with extended ASCII characters

### ISSUE THREAD

**Issue by gareth-j** [jupyterhub/binderhub#1045]
I've just encounterd an issue when using a DockerHub password such as `KÎRËTæJ6Æ5µóPD¦ëNýLhöN2ÂRYT«ã«¢s` created by my password manager. I was getting an Internal Server Error when trying to launch a repo. The binderhub pod log describing the error is below.

```
[E 200130 12:00:42 web:1621] Uncaught exception GET /build/gh/binder-examples/setup.py/master (10.244.2.1)
    HTTPServerRequest(protocol='https', host='<our-host>', method='GET', uri='/build/gh/binder-examples/setup.py/master', version='HTTP/1.1', remote_ip='10.244.2.1')
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/site-packages/tornado/web.py", line 1543, in _execute
        result = yield result
      File "/usr/local/lib/python3.6/site-packages/tornado/gen.py", line 1099, in run
        value = future.result()
      File "/usr/local/lib/python3.6/site-packages/binderhub/builder.py", line 293, in get
        image_manifest = await self.registry.get_image_manifest(*'/'.join(image_name.split('/')[-2:]).split(':', 1))
      File "/usr/local/lib/python3.6/site-packages/tornado/gen.py", line 1107, in run
        yielded = self.gen.throw(*exc_info)
      File "/usr/local/lib/python3.6/site-packages/binderhub/registry.py", line 199, in get_image_manifest
        auth_resp = yield client.fetch(auth_req)
      File "/usr/local/lib/python3.6/site-packages/tornado/gen.py", line 1099, in run
        value = future.result()
      File "/usr/local/lib/python3.6/site-packages/tornado/curl_httpclient.py", line 219, in _process_queue
        curl.info["headers"])
      File "/usr/local/lib/python3.6/site-packages/tornado/curl_httpclient.py", line 453, in _curl_setup_request
        curl.setopt(pycurl.USERPWD, native_str(userpwd))
    UnicodeEncodeError: 'ascii' codec can't encode characters in position 5-6: ordinal not in range(128)
```

**Comment by manics** [comment]
What version of tornado do you have?
It looks like this tornado PR is relevant: https://github.com/tornadoweb/tornado/pull/2397/files
In particular this line https://github.com/tornadoweb/tornado/pull/2397/files#diff-c95c2922049414ab5d74bd77ee0d2843L453
is appears in your error.

**Comment by gareth-j** [comment]
Could this not be fixed by updating the version of tornado in (`requirements.txt`)[https://github.com/jupyterhub/binderhub/blob/master/helm-chart/images/binderhub/requirements.txt]?

**Comment by betatim** [comment]
We now use tornado 6.0. could you let us know if the problem has been fixed by this upgrade or not?

**Comment by gareth-j** [comment]
Hi @betatim I can confirm that this is now fixed with the latest version (binderhub-0.2.0-n132.h1a8ce62). Thanks
